services/dataAccessLayer.js

```diff
@@ -47,6 +48,9 @@ const retrieveUsers = async ({
       users.push(user);
     });
     return { users, nextId, prevId };
+  } else if (discordId !== null) {
+    result = await userQuery.fetchUser({ discordId });
+    return result;
```

Comment:
Please filter the result according to the levels here
Do not directly return the result